### PR TITLE
Implement Claude Anthropic client with conversation history management

### PR DESCRIPTION
**Pull request by B-Naoufal**
- Added ClaudeClient struct to manage interactions with the Claude API.
- Implemented methods to send prompts, add messages to conversation history, and clear conversation history.
- Ensured that conversation history is maintained across multiple interactions with the API.
- Updated the Client interface to include methods for managing conversation history.

### CODE REVIEW

File: finowl-backend/pkg/ai/client_interface.go
Comment (finowl-near) [inline finowl-backend/pkg/ai/client_interface.go:1]: what's the purpose of this interface ?

File: finowl-backend/pkg/ai/client.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+func NewClaudeClient(apiKey string, modelName string, maxTokens int, streamEnabled, debug bool) (*ClaudeClient, error) {
+	if apiKey == "" {
+		return nil, errors.New("API key cannot be empty")
+	}
+
+	client := anthropic.NewClient(apiKey)
+
+	return &ClaudeClient{
+		client:        client,
+		modelName:     modelName,
```
Comment (finowl-near) [inline finowl-backend/pkg/ai/client.go:36]: maybe we should always use the same model ?

**Review by finowl-near — CHANGES_REQUESTED** [review]
you want to merge only client implementation and do another PR for integration with tweets ?

File: finowl-backend/pkg/ai/client.go (hunk 1 above)
Comment (B-Naoufal) [reply finowl-backend/pkg/ai/client.go:36]: later we will have a config file with all  needed configurations , but for now let's keep the client generic

File: finowl-backend/pkg/ai/client_interface.go
Comment (B-Naoufal) [reply finowl-backend/pkg/ai/client_interface.go:1]: unit tests later using mockgen

**Review by finowl-near — APPROVED** [review]

**Pull request merged**